### DISCUSSION TITLE
Github REST API: in_progress status is returning no workflow runs when there are runs in progress

### DISCUSSION THREAD

**Discussion by jmuollo** [community/community#53266] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Presently when using the github REST API to query workflow runs in the `in_progress` status (as per these [docs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository)) no runs are returned when there are runs in progress.
I first noticed this issue today at Wed, 19 Apr 2023 22:24:30 GMT.
This was working before today.

For example: 
https://api.github.com/repos/my-org/my-repo/actions/workflows/my-flow.yml/runs?status=in_progress
is returning:
```
{
    "total_count": 0,
    "workflow_runs": []
}
```

When at the same time this query:
https://api.github.com/repos/my-org/my-repo/actions/workflows/my-flow.yml/runs
is returning:
```
{
    "total_count": 38,
    "workflow_runs": [
        {
            "id": 4749561462,
            "name": "My flow",
            "node_id": "WFR_kwLOHatpMs8AAAABGxiOdg",
            "head_branch": "my-branch",
            "head_sha": "ce73ae7d98a7fb6bd417e3cfb2457e5779f2822d",
            "path": ".github/workflows/my-flow.yml",
            "display_title": "My flow",
            "run_number": 38,
            "event": "workflow_dispatch",
            "status": "in_progress",
            "conclusion": null,
            "workflow_id": 54271492,
            "check_suite_id": 12356913898,
            "check_suite_node_id": "CS_kwDOHatpMs8AAAAC4IeK6g",
            "url": "https://api.github.com/repos/my-org/my-repo/actions/runs/4749561462",
            "html_url": "https://github.com/my-org/my-repo/actions/runs/4749561462",
            "pull_requests": [],
            "created_at": "2023-04-20T01:33:47Z",
            "updated_at": "2023-04-20T01:33:47Z",
<snipped>
```

The first query should be returning the run with id 4749561462.

**Comment by jmuollo** [comment]
OK - it's working again now (`in_progress` status runs are now being returned). But it was definitely broken for several hours today.

**Comment by wolf4ood** [comment]
@jmuollo 

I have exactly the same issue with this now, it's still not working for me.  Did you change something or it just started working again?

**Reply by jmuollo** [reply]
It fixed itself, but it's back again now, so I've re-opened this discussion.

**Comment by paulagmf** [comment]
I have exactly the same issue with this now.

**Comment by jmuollo** [comment]
Yes this is happening again now.

**Comment by ZachOldham** [comment]
We are encountering the same issue but for the 'queued' and 'in_progress' status. It seems that their filtering is broken. This happened last year and it took them a month to fix, hopefully it gets fixed a lot faster this time.

**Reply by lpulley** [reply]
Yeah, it's definitely bigger than just `in_progress`. We have several runs at any given moment that are not appearing in the filter for their current status. This is also apparent on the web UI, on the Actions page; for example, there are queued runs that disappear when filtering to `status:queued`.

This is breaking our ephemeral CI system and I haven't received any answer yet on the ticket I opened with GitHub.

**Comment by mt2h** [comment]
I just ran this: workflow.get_runs(branch=branch, status="in_progress").get_page(-1)

The thing is that they changed the pagination order.

**Reply by lpulley** [reply]
I don't think this is all there is to it. Our workflow used to iterate through all of `get_workflow_runs(status=...)` and would simply never find certain runs.

I confirmed with Mohan in a private ticket that some runs are experiencing this problem, and they claimed that the engineering team has started some kind of job that is meant to resolve this. We'll see.

**Comment by dreamcoin1998** [comment]
I tested it and found that it is able to retrieve a task with the first workflow status of "in_progress", but it is unable to retrieve a task with the same status from a second workflow.

**Comment by jmuollo** [comment]
It's happening again today.

**Comment by Lawouach** [comment]
I echo that. I was really confused about it as well.

**Comment by goliaro** [comment]
I've also been experiencing similar issues in the past couple of weeks, and today in particular.

**Comment by tomgrossman** [comment]
same here today 😑

**Comment by sigurdmwahl** [comment]
Same here, I see other statuses returning but not in_progress. Any update from GH here?

**Comment by rezaghz** [comment]
same here today

**Comment by ghost** [comment]
Still facing the same issue, no ETA yet :(

**Comment by alexanderkranga** [comment]
Happens to us as well. We are running self-hosted GitHub runners with Kubernetes and that solution relies on this API. Because of this issue our runners are not being scaled up when needed. This issue must be prioritized

**Comment by MrMint** [comment]
We are also running into this issue. It is affecting both the scaling of our self-hosted runners and workflows using octokit to monitor in_progress workflows for CD purposes.

**Comment by ZachOldham** [comment]
We have a support ticket open with github about this, they have on multiple occasions told us the issue is fixed, which is then immediately proven false the first time we test it. I have no idea what is going on over there.
